clang/docs/ReleaseNotes.rst

```diff
@@ -137,6 +137,12 @@ Removed Compiler Flags
 
 Attribute Changes in Clang
 --------------------------
+- Introduced a new function attribute ``__attribute__((amdgpu_max_num_work_groups(x, y, z)))`` or
```

Comment:
s/work_groups/workgroup/

Reply:
There are existing attributes that have workgroup spelled as two separate words: [flat-work-group-size](https://clang.llvm.org/docs/AttributeReference.html#amdgpu-flat-work-group-size), [reqd_work_group_size](https://docs.xilinx.com/r/2021.1-English/ug1393-vitis-application-acceleration/reqd_work_group_size).
Pls let me know if you still want workgroup as one word.

Reply:
Ugh. The ISA manuals usually use "workgroup". read_work_group_size came from OpenCL. We made up amdgpu_flat_work_group_size

Reply:
The backend facing parts seem more consistently to be workgroup. e.g in the HSA metadata, we have .workgroup_size_hint, .reqd_workgroup_size. As horrible as it is, maybe it's best to keep it this way for the clang attribute, and rename all the backend bits to be "workgroup"?

Reply:
In the case of flat workgroup size, the LLVM attribute is called `amdgpu-flat-work-group-size`, but the metadata is `.max_flat_workgroup_size`. I suppose we can copy that and change the metadata from `.max_num_work_groups_x` to `.max_num_workgoups_x` and so on.

Do you want the LLVM attribute to be changed from `amdgpu-max-num-work-groups` to `amdgpu-max-num-workgroups` as well? Note that in the file `AMDGPUUsage.rst` the word `work-group` with a hyphen is used a lot.

Reply:
I think some of the AMDGPUUsage work-groups are actually wrong based on the actual code.

Reply:
I think ".max_num_workgroups" "amdgpu-max-num-workgroups" and "amdgpu_max_num_work_groups" is the most consistent with the existing uses